sfxclient/httpsink.go

```diff
@@ -131,6 +135,24 @@ func toMT(mt datapoint.MetricType) com_signalfx_metrics_protobuf.MetricType {
 	panic(fmt.Sprintf("Unknown metric type: %d\n", mt))
 }
 
+var toECMap = map[event.Category]com_signalfx_metrics_protobuf.EventCategory{
+	event.USERDEFINED:      com_signalfx_metrics_protobuf.EventCategory_USER_DEFINED,
+	event.ALERT:            com_signalfx_metrics_protobuf.EventCategory_ALERT,
+	event.AUDIT:            com_signalfx_metrics_protobuf.EventCategory_AUDIT,
+	event.JOB:              com_signalfx_metrics_protobuf.EventCategory_JOB,
+	event.COLLECTD:         com_signalfx_metrics_protobuf.EventCategory_COLLECTD,
+	event.SERVICEDISCOVERY: com_signalfx_metrics_protobuf.EventCategory_SERVICE_DISCOVERY,
+	event.EXCEPTION:        com_signalfx_metrics_protobuf.EventCategory_EXCEPTION,
+}
+
+func toEC(ec event.Category) com_signalfx_metrics_protobuf.EventCategory {
```

Comment:
you already assigned them to the same values above, why not just use proto.EnumName(com_signalfx_metrics_protobuf .EventCategory_name, int32(theid)) instead of this whole toECMap and toEC() method in the one place toEC() is used. The reason datapoints used this overly verbose map is that they used iota for constants. I'm leaning more and more towards just putting the protobuf enum in event.go and forcing client to do the conversion from string since they already need to change.